botocore/utils.py

```diff
@@ -1569,6 +1571,45 @@ def set_endpoint(self, request, **kwargs):
     def _use_accesspoint_endpoint(self, request):
         return 's3_accesspoint' in request.context
 
+    def _validate_fips_supported(self, request):
+        client_region = request.context.get('client_region', '')
```

Comment:
Is there a particular reason why region is pulled from the request instead of use the `self._region` property? It seems like a fair amount of the logic here uses that for the client region.